File: niche_difer_sp.R
niche_difer_sp<-function(sp_chr,T){
  source("niche_difer2.R")
  n<-dim(sp_chr)
  y2<-matrix(nrow = n[length(n)],ncol = n[length(n)])
  
  if (length(n)==2) {
    for (i in 1:(n[length(n)]-1)) {
      for (j in ((i+1):n[length(n)])) {
        y2[i,j]<-niche_difer2(as.matrix(sp_chr[,i]),as.matrix(sp_chr[,j]),T)
      }
    }
    
  }else{
    for (i in 1:(n[length(n)]-1)) {
      for (j in ((i+1):n[length(n)])) {
        y2[i,j]<-niche_difer2(as.matrix(sp_chr[,,i]),as.matrix(sp_chr[,,j]),T)
      }
    }
  }
  
  y2<-y2
}

File: niche_difer_sp.m
function y=niche_difer_sp(sp_chr,T)
% To use in 'combina_niche3.m'
% INPUTS:
%
% sp_chr: three dimensional matrices with the species chromatograms (alpha category by p environmental variables by species)
% T: the threshold of minimal abundance in a category for the niche breadth estimation
% 
% OUTPUTS:
%
% y: index D
%
% Grégory Beaugrand Avril 2021

[~,~,z]=size(sp_chr);
y=zeros(z,z)+nan;
for i=1:z
    for j=i+1:z
        y(i,j)=niche_difer2(squeeze(sp_chr(:,:,i)),squeeze(sp_chr(:,:,j)),T);
    end
end

